### PR TITLE
fix: Yarn PnP support

### PR DESCRIPTION
**Pull request by lihbr**
**Resolves**: https://community.prismic.io/t/unable-to-setup-prismic-with-yarn-pnp-mode/16218

### Description

Slice Machine doesn't work with Yarn PnP. This PR leverages Yarn PnP API when available to properly resolve module handled by Yarn PnP (well... I really don't enjoy the non-standard way PnP is doing things...)

The PR also explicitly declares `readable-web-to-node-stream` which is directly used by our bundled code and updates `@segment/analytics-node` to accommodate Yarn ghost dependencies requirement.

To get a hang of the changes, I've created this "minimal reproduction/fix" of sort highlighting the issue. Refer to the README for how to operate it.

[pnp-poc.zip](https://github.com/user-attachments/files/16086339/pnp-poc.zip)

### Checklist

- [x] If my changes require tests, I added them.
   I don't think there's a good way to test that, I'll QA manually with the Alpha.
- [x] If my changes affect backward compatibility, it has been discussed.
- [x] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

1. Create a Next.js website, kill the process when it gets to install dependencies
   ```bash
   npx create-next-app
   ```
2. Get in the new app, enable Yarn Berry, and install dependencies with Yarn
   ```bash
   cd my-app && yarn set version berry && yarn
   ```
3. Init Slice Machine with latest, it should fail
   ```
   yarn dlx @slicemachine/init@latest
   ```
4. Init Slice Machine with alpha, it should work
   ```
   yarn dlx @slicemachine/init@alpha.lh-fix-yarn-pnp-support
   ```
5. Start Slice Machine, it should work
   ```
   yarn slicemachine
   ```

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

**Comment by lihbr** [comment]
WIP: https://prismic-team.slack.com/archives/C02L3FN3AJK/p1720022475787989

File: packages/manager/src/lib/requireResolve.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Resolves a module path with `createRequire().resolve()` with Yarn PnP
+ * support.
+ *
+ * @param id - Module to resolve.
+ * @param from - Location to resolve the module from.
+ *
+ * @returns - Resolved module path.
+ */
+export default (id: string, from: string): string => {
```
Comment (angeloashmore) [inline packages/manager/src/lib/requireResolve.ts]: ⚠️ #issue: Could we use a named export to be consistent with other files?

File: packages/manager/src/lib/requireResolve.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	let _id = id;
+
+	// Support Yarn PnP
+	if (
+		process.versions.pnp &&
+		"findPnpApi" in _module &&
+		_module.findPnpApi instanceof Function
```
Comment (angeloashmore) [inline packages/manager/src/lib/requireResolve.ts]: 💡 #idea: `typeof _module.findPnpApi === "function"` seems more idiomatic to me. What do you think? Is there a benefit to using `instanceof`?

File: packages/manager/src/lib/requireResolve.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * Resolves a module path with `createRequire().resolve()` with Yarn PnP
+ * support.
+ *
+ * @param id - Module to resolve.
+ * @param from - Location to resolve the module from.
+ *
+ * @returns - Resolved module path.
+ */
+export default (id: string, from: string): string => {
+	let _id = id;
```
Comment (angeloashmore) [inline packages/manager/src/lib/requireResolve.ts]: 💡 #idea: The `_` prefix usually denotes private data. In this case, `_id` is more like a temporary mutable value. How about naming it `resolvedID`?

**Review by xrutayisire — APPROVED** [review]
Super nice! Tested with yarn, yarn pnp, npm, let's go! 🚀

We are planning to release SM next week middle, end of the week. If you want to release before on your own to fix this bug, feel free to do it 🙂

File: packages/manager/src/lib/requireResolve.ts (hunk 1 above)
Comment (lihbr) [reply packages/manager/src/lib/requireResolve.ts]: Copilot held my hand too far 🫠

File: packages/manager/src/lib/requireResolve.ts (hunk 2 above)
Comment (lihbr) [reply packages/manager/src/lib/requireResolve.ts]: That also works, not sure why I went `instanceof` here

**Pull request merged**